packages/web3-providers/src/NFTScan/helpers/EVM.ts

```diff
@@ -134,16 +139,18 @@ export function createNonFungibleAsset(chainId: ChainId, asset: EVM.Asset): NonF
             chainId,
             schema,
             address: asset.contract_address,
-            name,
+            name: contractName,
             symbol,
         },
         collection: {
             chainId,
-            name,
+            name: contractName,
             slug: name,
             description,
             address: asset.contract_address,
-            iconURL: urlcat(NFTSCAN_LOGO_BASE + '/:id', { id: asset.contract_address + '.png' }),
+            iconURL: collectionContext
+                ? collectionContext.logo_url
```

Comment:
If `collectionContext.logo_url` is null, we will render a fallback logo instead directly.